### PR TITLE
Slf4j over ant

### PR DESCRIPTION
**Pull request by colezlaw**
Patch for jeremylong/DependencyCheck#164 and jeremylong/DependencyCheck#239. This switches everything to Log4J. The CLI still uses JULi as its implementation, but all the logging messages are going through SLF4J. The Maven plugin and the Ant plugin both use their own internal logging mechanisms rather than using SLF4J. Furthermore, adapters have been written for both so that SLF4J messages from Core are routed through the native logging interfaces for each.

### CODE REVIEW

File: dependency-check-cli/pom.xml [hunk 1]
```diff
@@ -322,5 +322,12 @@ Copyright (c) 2012 - Jeremy Long. All Rights Reserved.
             <artifactId>dependency-check-utils</artifactId>
             <version>${project.parent.version}</version>
         </dependency>
+        <!-- Logging implementation. We may change this in the future -->
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-jdk14</artifactId>
+            <version>1.7.12</version>
```
Comment (hansjoachim) [inline dependency-check-cli/pom.xml:329]: Please put version numbers for dependencies under dependencyManagement in the root pom. :) Makes it easier to keep track of them when they're in one place, and it's a single place to upgrade once a new release is out. (See also #252).

File: dependency-check-core/pom.xml [hunk 2]
```diff
@@ -362,6 +362,26 @@ Copyright (c) 2012 Jeremy Long. All Rights Reserved.
     <dependencies>
         <!-- Note, to stay compatible with Jenkins installations only JARs compiled to 1.6 can be used -->
         <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>${slf4j.version}</version>
```
Comment (hansjoachim) [inline dependency-check-core/pom.xml:367]: I might be missing something here, but where is the slf4j.version property defined?

**Comment by hansjoachim** [comment]
Hi :)

I'm just an occasional contributor, so I haven't looked at this in-depth, but I left a couple of comments.

**Comment by colezlaw** [comment]
Closing this to rebase with merges and fix issues with versioning.

**Comment by colezlaw** [comment]
@hansjoachim I closed the pull request to fix your questions plus another Jeremy had open. However, I've always been puzzled by the dependencyManagement thing - I know how it's SUPPOSED to work, but if I define a version for slf4j-api in the parent POM dependencyManagement section, then define the dependency without a version in child projects, I still get an error in the child that I didn't specify a version. However, things like commons-io don't have version numbers and those work.

**Comment by hansjoachim** [comment]
Like you said, that sounds like it should work. Even stranger if it works for other dependencies. I don't know what would be causing that, but if you have an example, let me know and I can take a look at it. :)

**Comment by colezlaw** [comment]
Sorry. I misspelled "org". That's fixed.

What all dependency versions do you want in dependencyManagement? All of
them? Or just ones that appear in multiple projects?

**Comment by hansjoachim** [comment]
> What all dependency versions do you want in dependencyManagement? All of
> them? Or just ones that appear in multiple projects?

Please add all of them to dependencyManagement. For one thing they might be picked up by other modules in the future (or when new ones are introduced) and it means there will be a single file which needs updating when newer versions are released. :)

Btw, I don't know how obvious it is, but I've tried to keep the list in dependencyManagment in alphabetical order.

**Pull request closed without merging**